ecl/hthor/hthor.cpp

```diff
@@ -9553,7 +9558,9 @@ bool CHThorCsvReadActivity::openNext()
         while (lines-- && !inputstream->eos())
         {
             size32_t numAvailable;
-            const void * next = inputstream->peek(maxDiskSize, numAvailable);
+            // Probably the first fixing point. This changes use same value as Thor does
+            // to skip CSV header. 
+            const void * next = inputstream->peek(maxRowSize, numAvailable);
             inputstream->skip(csvSplitter.splitLine(numAvailable, (const byte *)next));
```

Comment:
This is not the right call to make:

This should use the same scheme as thor calling splitLine(steam, max) which dynamically increases the size requetested until the max is reached. Rather than peek on the maximum line length.